mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/dao/PostgresMailboxDAO.java

```diff
@@ -51,6 +60,19 @@
 public class PostgresMailboxDAO {
     private static final char SQL_WILDCARD_CHAR = '%';
     private static final String DUPLICATE_VIOLATION_MESSAGE = "duplicate key value violates unique constraint";
+    private static final Function<MailboxACL, Hstore> MAILBOX_ACL_TO_HSTORE_FUNCTION = acl -> Hstore.hstore(acl.getEntries()
+        .entrySet()
+        .stream()
+        .collect(Collectors.toMap(
+            entry -> entry.getKey().serialize(),
+            entry -> entry.getValue().serialize())));
+
+    private static final Function<Hstore, MailboxACL> HSTORE_TO_MAILBOX_ACL_FUNCTION = hstore -> new MailboxACL(hstore.data()
+        .entrySet()
+        .stream()
+        .collect(Collectors.toMap(
+            entry -> MailboxACL.EntryKey.deserialize(entry.getKey()),
+            Throwing.function(entry -> MailboxACL.Rfc4314Rights.deserialize(entry.getValue())))));
```

Comment:
Please error log + filter out invalid entries. Not sure we want a hard crash preventing to read emails if we have a serialisation glitch here...